playwright/e2e/crypto/crypto.spec.ts

```diff
@@ -599,5 +601,124 @@ test.describe("Cryptography", function () {
+                        await encryptionStatePromise;
+
+                        await client.sendTextMessage(roomId, "This should be undecryptable");
+
+                        await client.invite(roomId, alice.userId);
+
+                        return roomId;
+                    },
+                    { alice },
+                );
+
+                // Alice accepts the invite
+                await expect(
+                    page.getByRole("group", { name: "Invites" }).locator(".mx_RoomSublist_tiles").getByRole("treeitem"),
+                ).toHaveCount(1);
+                await page.getByRole("treeitem", { name: "Test room" }).click();
+                await page.locator(".mx_RoomView").getByRole("button", { name: "Accept" }).click();
+
+                // Bob sends an encrypted event and an undecryptable event
+                await bob.evaluate(
+                    async (client, { roomId }) => {
+                        await client.sendTextMessage(roomId, "This should be decryptable");
+                        const { event_id: lastEventId } = await client.sendEvent(
+                            roomId,
+                            "m.room.encrypted" as any,
+                            {
+                                algorithm: "m.megolm.v1.aes-sha2",
+                                ciphertext: "this+message+will+be+undecryptable",
+                                device_id: client.getDeviceId()!,
+                                sender_key: (await client.getCrypto()!.getOwnDeviceKeys()).ed25519,
+                                session_id: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+                            } as any,
+                        );
+                    },
+                    { roomId },
+                );
+
+                // We wait for the event tiles that we expect from the messages that
+                // Bob sent, in sequence.
+                await expect(
+                    page.locator(`.mx_EventTile`).getByText("You don't have access to this message"),
+                ).toBeVisible();
+                await expect(
+                    page.locator(`.mx_EventTile`).getByText("This should be decryptable"),
+                ).toBeVisible();
+                await expect(
+                    page.locator(`.mx_EventTile`).getByText("Unable to decrypt message"),
+                ).toBeVisible();
+
+                // And then we ensure that they are where we expect them to be
+                // Alice should see these event tiles:
+                // - first message sent by Bob (undecryptable)
+                // - Bob invited Alice
+                // - Alice joined the room
+                // - second message sent by Bob (decryptable)
+                // - third message sent by Bob (undecryptable)
+                const tiles = await page.locator(".mx_EventTile").all();
+                expect(tiles.length).toBeGreaterThanOrEqual(5);
```

Comment:
OOI when will there be more than 5 tiles?

Reply:
It uses an EventTile for things like the thing that indicates that encryption is enabled.  By the time we get to this point in the test, that tile will have disappeared.  I'm not sure if other previous events will get a tile (I think they all get thrown in a group).  But basically, by checking that the length is at least 5, if other bits of the react-sdk change so that other tiles are shown, this test won't immediately break.